web/testing/build-visual-keyboard/index.html

```diff
@@ -50,56 +51,103 @@
                 platform == 'phone' ? 240 : 360;
       };
 
+      let shadowfy = function(container, style) {
+        const docBase = container.children[0];
+        container.removeChild(docBase);
+
+        const shadow = container.attachShadow({ mode: "open" });
+        shadow.appendChild(docBase);
+
+        const baseOSKStyle = document.createElement('link');
+        baseOSKStyle.href = "../../release/unminified/web/osk/kmwosk.css";
+        baseOSKStyle.rel="stylesheet";
+        baseOSKStyle.type="text/css";
+        shadow.appendChild(baseOSKStyle);
+
+        shadow.appendChild(style);
+      }
+
+      let documentKeyboard = async function(kbdid, langid, formFactor, layer) {
+        let docTarget = document.getElementById("docTarget");
+        label = document.createElement('h3');
+        docTarget.appendChild(label);
+
+        let idLabel = document.createElement('code');
+        docTarget.appendChild(idLabel);
+
+        const container = document.createElement('div');
+
+        await kmw.addKeyboards(`${kbdid}@${langid}`);
+        await kmw.setActiveKeyboard(kbdid, langid);
+        docBase = keyman.BuildVisualKeyboard(kbdid, 1, formFactor, layer);
+
+        label.textContent = `${keyman.getKeyboard(kbdid).Name} - ${formFactor} - ${layer} layer`;
+        idLabel.textContent = `${kbdid.replace(/^Keyboard_/)}`;
+        container.appendChild(docBase);
+
+        docTarget.appendChild(container);
+
+        await Promise.resolve(); // Trigger message queue:  let the mutation observer do its thing.
+
+        // Grab the stylesheet element that was inserted; there should be no intervening additions,
```

Comment:
> there should be no intervening additions

This may be true for this page, but not generally true. Anything can happen when we `await` :grin: